### PR TITLE
Add live reload for static sites

### LINKED ISSUE #653: preview --watch does not work for site
We need to watch the specified `bucket` directory for changes and sync with the preview namespace on changes

edit from @EverlastingBugstopper

### PR DESCRIPTION
**Pull request by gabbifish**
Fixes #653. This PR adds support for `wrangler preview --watch` to work with static assets.

There is still a quirks I want to work out in the future: is the 2000 ms cooldown period enough to account for large static site generator builds that exceed that time? (I think it should be, we can wait to hear otherwise...)

### CODE REVIEW

**Comment by EverlastingBugstopper** [comment]
Fixes #653

**Comment by ashleymichal** [comment]
QA-ing this; the output confuses me, esp the duplicate "ignoring" line, and the "Success" message, i'm not sure what has succeeded in each case.


``` sh
$ wrangler-dev preview --watch
🌀  Creating namespace for Workers Site "__hello-react-workers_sites_assets_preview"
💁  ignoring directory node_modules
✨  Success
💁  ignoring directory node_modules
✨  Success
✨  Built successfully, built project size is 4 KiB.
```

File: src/commands/build/wranglerjs/mod.rs [hunk 1]
```diff
@@ -58,6 +58,8 @@ pub fn run_build_and_watch(target: &Target, tx: Option<Sender<()>>) -> Result<()
     let (mut command, temp_file, bundle) = setup_build(target)?;
     command.arg("--watch=1");
 
+    let is_site = target.site.clone();
```
Comment (xtuc) [inline src/commands/build/wranglerjs/mod.rs]: This should be maybe_site I think

File: src/commands/build/wranglerjs/mod.rs [hunk 2]
```diff
@@ -68,9 +70,21 @@ pub fn run_build_and_watch(target: &Target, tx: Option<Sender<()>>) -> Result<()
         let mut watcher = notify::watcher(watcher_tx, Duration::from_secs(1))?;
 
         watcher.watch(&temp_file, RecursiveMode::Recursive)?;
-
         info!("watching temp file {:?}", &temp_file);
 
+        if let Some(site) = is_site {
```
Comment (xtuc) [inline src/commands/build/wranglerjs/mod.rs]: Similar to above, this could be:
```rs
                if let Some(site_config) = target.site.clone() {

```

**Comment by gabbifish** [comment]
@ashleymichal ugh something in master is responsible for these double printouts, they did not happen until I merged master in. I'm investigating.

File: src/commands/build/wranglerjs/mod.rs (hunk 1 above)
Comment (ashleygwilliams) [reply src/commands/build/wranglerjs/mod.rs]: why maybe site? i think is_site says if it is a site or not, the maybe will be very tough to understand later. if target.site exists, it's a site. am i missing something perhaps?

File: src/commands/build/wranglerjs/mod.rs (hunk 2 above)
Comment (ashleymichal) [reply src/commands/build/wranglerjs/mod.rs]: agreed but not super critical.

**Pull request merged**